### ISSUE TITLE
Migrating to Kendo Theme Bootstrap 8.0.1

### ISSUE THREAD

**Issue by Schoof-T** [telerik/kendo-themes#5090]
**Describe the issue or request**
We have been using the Kendo-theme-bootstrap, together with bootstrap (5.1.3 currently) for Blazor for a couple of years. 
This week we wanted to upgrade to the latest version of Kendo UI Bootstrap, but we are noticing the colours are no longer matching with our bootstrap.

I noticed you are also no longer dependent on bootstrap, but in our applications we use mostly bootstrap classes and component together with more advanced Telerik components. Is this no longer supported?

Is there a migration guide, which variables need to be overwritten?

**Additional context**
The way we have set up our theme is as follows:
- Imported both the scss of kento UI bootstrap and bootstrap
- Created our very own variables file, with our colours
- Created a scss file for bootstrap where we overwrite the variables
- Created a scss file for telerik bootstrap where we overwrite the variables
The telerik bootstrap scss file we created looks as following:
```
@import '../_variables.scss';

$border-radius: 0.35rem;
$primary: $jdn-red;
$secondary: #6c757d; // $secondary from bootstrap
$info: $jdn-light;
$success: $jdn-green;
$warning: $jdn-yellow;
$error: $jdn-red;
$body-text: #212529;
$body-bg: #ffffff;
$headings-text: #212529;
$subtle-text: #6c757d;
$disabled-text: #8f8f8f;
$component-text: #212529;
$component-bg: #ffffff;
$base-text: #212529;
$base-bg: #f7f7f7;
$hovered-text: #16181b;
$hovered-bg: #e9ecef;
$selected-text: #ffffff;
$selected-bg: $jdn-red;
$button-text: #212529;
$button-bg: #e4e7eb;
$link-text: $jdn-font-color;
$link-hover-text: shift-color($jdn-red, 20%);
$series-a: #0275d8;
$series-b: #5bc0de;
$series-c: #5cb85c;
$series-d: #f0ad4e;
$series-e: #e67d4a;
$series-f: #d9534f;
$font-size-base: $jdn-font-size-base;


// Remove styling of valid inputs
$kendo-input-valid-border: #ced4da;
$kendo-input-valid-shadow: 0 0 0 0.25rem rgba(186,12,47,.25);

// grid
$table-cell-padding: 0.35rem;

// loader
$loader-container-panel-bg: initial;
$loader-container-panel-border-style: initial;

@import "../../lib/telerik/dist/all.scss";
```

Thank you in advance.

**Comment by Schoof-T** [comment]
I figured out that I have to overwrite the colors as follows:

```
$kendo-colors: ( 
primary: $primary, 
secondary: #6c757d, 
info: $jdn-light,
success: $jdn-green,
warning: $jdn-yellow,
danger: $jdn-red,
);
```

But this doesn't change the colour of a lot of other items. Such as the :hovers, the :actives, grid paging, toasters...

**Comment by epetrow** [comment]
Hey, @Schoof-T with version [8.0.0](https://github.com/telerik/kendo-themes/releases/tag/v8.0.0) of the Kendo Themes we have enabled the new Color system which introduces a new way of customizing the old color variables like `$kendo-color-primary, $kendo-color-secondary, $kendo-base-bg, etc.`

You can find detailed information about the Color system [here](https://www.telerik.com/design-system/docs/foundation/color/) and how to upgrade to Kendo Theme Bootstrap v8.0.1 [here](https://www.telerik.com/design-system/docs/themes/theme-bootstrap/migration/).

The gist of it is:

- Want to opt-out and preserve the old behaviour? - Simply set the  `$kendo-enabled-color-system` variable to `false`
- If you wish to migrate and make use of the new Color System - Follow the [migration guide for Bootstrap](https://www.telerik.com/design-system/docs/themes/theme-bootstrap/migration/)

Basically this is how customizing the Kendo Theme Bootstrap colors looks like as of v8.0.0

```scss
@import "@progress/kendo-theme-core/scss/functions/index.import.scss";

$kendo-colors: ();
$kendo-colors: k-map-merge($kendo-colors, (
app-surface: #f8f8f8,
on-app-surface: #111111,
subtle: rgba(0, 0, 0, 0.54),
surface: #f3f2f1,
surface-alt: #fafafa,
border: rgba(0, 0, 0, 0.2),
border-alt: rgba(0, 0, 0, 0.3)
),
k-generate-color-variations( 'primary', #9c27b0, 'bootstrap' ),
k-generate-color-variations( 'secondary', #e51a5f, 'bootstrap' ),
k-generate-color-variations( 'info', #0058e9, 'bootstrap' ),
k-generate-color-variations( 'success', #37b400, 'bootstrap' ),
k-generate-color-variations( 'base', #eaeaea, 'bootstrap' )
);

@import "@progress/kendo-theme-bootstrap/dist/all.scss"
```

**Comment by Schoof-T** [comment]
Thanks! I couldn't find the documentation.

I'm trying to import "@progress/kendo-theme-core/scss/functions/index.import.scss"
But the file itself again contains the same import?
![image](https://github.com/telerik/kendo-themes/assets/2319986/a7b5f1dc-56b5-4ceb-ad8f-3b97262d77bb)

So I can't import it. Any idea how to get around that?

**Comment by epetrow** [comment]
@Schoof-T, I am not sure if it became clear from my previous comment, but you would only need to import `"@progress/kendo-theme-core/scss/functions/index.import.scss";` in the same file where you customize the Kendo Bootstrap theme.

Just make sure it is before any of the customizations to the colors.

**Comment by Schoof-T** [comment]
> @Schoof-T, I am not sure if it became clear from my previous comment, but you would only need to import `"@progress/kendo-theme-core/scss/functions/index.import.scss";` in the same file where you customize the Kendo Bootstrap theme.
> 
> Just make sure it is before any of the customizations to the colors.

Yeah, I made my overrides file look like this:

```
@import "../_variables.scss";
@import "../../lib/telerik/scss/core/functions/index.import.scss";

$border-radius: 0.35rem;

$kendo-colors: ();
$kendo-colors: k-map-merge($kendo-colors,  
    k-generate-color-variations( 'primary', $jdn-red, 'bootstrap' ),
    k-generate-color-variations( 'secondary', #6c757d, 'bootstrap' ),
	k-generate-color-variations( 'info', $jdn-light, 'bootstrap' ), 
	k-generate-color-variations( 'success', $jdn-green, 'bootstrap' ), 	
	k-generate-color-variations( 'warning', $jdn-yellow, 'bootstrap' ), 
	k-generate-color-variations( 'danger', $jdn-red 'bootstrap' ) 
);

$body-text: #212529;
$body-bg: #ffffff;
$headings-text: #212529;
$subtle-text: #6c757d;
$disabled-text: #8f8f8f;
$component-text: #212529;
$component-bg: #ffffff;
$base-text: #212529;
$base-bg: #f7f7f7;
$hovered-text: #16181b;
$hovered-bg: #e9ecef;
$selected-text: #ffffff;
$selected-bg: $jdn-red;
$button-text: #212529;
$button-bg: #e4e7eb;
$link-text: $jdn-font-color;
$link-hover-text: shift-color($jdn-red, 20%);
$series-a: #0275d8;
$series-b: #5bc0de;
$series-c: #5cb85c;
$series-d: #f0ad4e;
$series-e: #e67d4a;
$series-f: #d9534f;
$font-size-base: $jdn-font-size-base;


// Remove styling of valid inputs
$kendo-input-valid-border: #ced4da;
$kendo-input-valid-shadow: 0 0 0 0.25rem rgba(186,12,47,.25);

// grid
$table-cell-padding: 0.35rem;

// loader
$loader-container-panel-bg: initial;
$loader-container-panel-border-style: initial;

@import "../../lib/telerik/dist/all.scss";

// Import overrides
@import 'jdn.telerik.overrides.scss';
```

But upon compiling I get the following error:

```
Error		Error: Can't find stylesheet to import.
  ╷
1 │ @import "@progress/kendo-theme-core/scss/functions/index.import.scss";
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
```

So I went to look at the index.import.scss file, and I found that the content of said file is the following:

`@import "@progress/kendo-theme-core/scss/functions/index.import.scss";`

Which will not compile on my system, because I don't have a folder that starts with @progress. 

![image](https://github.com/telerik/kendo-themes/assets/2319986/d594c8a4-ea69-4c73-9e45-1fd612df94cd)

**Comment by epetrow** [comment]
> > @Schoof-T, I am not sure if it became clear from my previous comment, but you would only need to import `"@progress/kendo-theme-core/scss/functions/index.import.scss";` in the same file where you customize the Kendo Bootstrap theme.
> > Just make sure it is before any of the customizations to the colors.
> 
> Yeah, I made my overrides file look like this:
> 
> ```
> @import "../_variables.scss";
> @import "../../lib/telerik/scss/core/functions/index.import.scss";
> 
> $border-radius: 0.35rem;
> 
> $kendo-colors: ();
> $kendo-colors: k-map-merge($kendo-colors,  
>     k-generate-color-variations( 'primary', $jdn-red, 'bootstrap' ),
>     k-generate-color-variations( 'secondary', #6c757d, 'bootstrap' ),
> 	k-generate-color-variations( 'info', $jdn-light, 'bootstrap' ), 
> 	k-generate-color-variations( 'success', $jdn-green, 'bootstrap' ), 	
> 	k-generate-color-variations( 'warning', $jdn-yellow, 'bootstrap' ), 
> 	k-generate-color-variations( 'danger', $jdn-red 'bootstrap' ) 
> );
> 
> ...
> 
> @import "../../lib/telerik/dist/all.scss";
> ```
> 
> But upon compiling I get the following error:
> 
> ```
> Error		Error: Can't find stylesheet to import.
>   ╷
> 1 │ @import "@progress/kendo-theme-core/scss/functions/index.import.scss";
>   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   ╵
> ```
> 
> So I went to look at the index.import.scss file, and I found that the content of said file is the following:
> 
> `@import "@progress/kendo-theme-core/scss/functions/index.import.scss";`
> 
> Which will not compile on my system, because I don't have a folder that starts with @progress.
> 
> We import the files from jsDelivr: https://www.jsdelivr.com/package/npm/@progress/kendo-theme-bootstrap

Is it possible to share a sample demo, so I can inspect it?

**Comment by Schoof-T** [comment]
@epetrow 

Please find the sample attached.

[TelerikUI.zip](https://github.com/user-attachments/files/16025900/TelerikUI.zip)

**Comment by Schoof-T** [comment]
Hi @epetrow, any update on this?

**Comment by epetrow** [comment]
In order to compile the Kendo Bootstrap theme the following changes were made:

libman.json

```json

{
  "version": "1.0",
  "libraries": [
    {
      "provider": "jsdelivr",
      "library": "@progress/kendo-theme-core@8.0.1",
      "destination": "wwwroot/lib/@progress/kendo-theme-core"
    },
    {
      "provider": "jsdelivr",
      "library": "@progress/kendo-theme-bootstrap@8.0.1",
      "destination": "wwwroot/lib/@progress/kendo-theme-bootstrap/"
    }
  ]
}
```

compilerconfig.json

```json

[
  {
    "inputFile": "content/custom/telerik/custom-telerik.scss",
    "outputFile": "content/custom/telerik/custom-telerik.css",
    "options": {
      "loadPaths": "wwwroot/lib/"
    }
  }
]
```

I've also attached a working example [MigrateBootstrap.zip](https://github.com/user-attachments/files/16263776/MigrateBootstrap.zip)

**Comment by molequetu** [comment]
Shouldn't this at least be mentioned in the release notes;;; 
We cannot always trying to find what is going wrong in every upgrade.

**Comment by molequetu** [comment]
@epetrow I am commenting here, so I don't open a new issue
As described in migration guide, some sass variables have been replaced with others. For example, 
$kendo-disabled-text and $kendo-body-text are now as on-app-surface.

Migrating as described using this strategy, we cannot have different colors as we had for these variables.
Now in our sass file we have a structure as of
```
$kendo-body-text: #212529; // on-app-surface
$kendo-disabled-text: #8f8f8f; // on-app-surface
$kendo-component-text: #212529; // on-app-surface
$kendo-base-text: #212529; // on-app-surface
$kendo-hover-text: #16181b; // on-app-surface
$kendo-body-bg: #ffffff; // app-suface
$kendo-hover-bg: #e9ecef; // base-subtle-hover
$kendo-selected-text: #ffffff; // on-primary
$kendo-subtle-text: #6c757d; // subtle
$kendo-component-bg: #ffffff;  // surface-alt
$kendo-base-bg: #f8f9fa;  // surface
$headings-text: #212529; // not available
```
As you can see, we used different colors for these variables(e.g $kendo-disabled-text). If we use the strategy of migration, implemented as follows

```
// custom design
@import "../../athena/scss/base/_variables.scss";
@import "@progress/kendo-theme-core/scss/functions/index.import.scss";

$kendo-colors: ();
$kendo-colors: k-map-merge($kendo-colors,
(
app-surface: #ffffff,
on-app-surface: #212529,
subtle: #6c757d,
surface: #f8f9fa,
surface-alt: #ffffff,
base-subtle-hover: #e9ecef,
on-primary: #ffffff
),
k-generate-color-variations( 'primary', $primary, 'bootstrap' ),
k-generate-color-variations( 'secondary', $secondary, 'bootstrap' ),
k-generate-color-variations( 'info', $info, 'bootstrap' ),
k-generate-color-variations( 'success', $success, 'bootstrap' ),
k-generate-color-variations( 'warning', #ffc107, 'bootstrap' ),
k-generate-color-variations( 'error',  #dc3545, 'bootstrap' )
);
```
the generated css, does not include none color for -k.disabled classes, neither the new one as of on-app-surface 
Css generated after migration
```
.k-disabled,
.k-widget[disabled],
.k-disabled {
  outline: none;
  cursor: default;
  color: var(--kendo-disabled-text, );
  background-color: var(--kendo-disabled-bg, );
  border-color: var(--kendo-disabled-border, );
  opacity: var(--kendo-disabled-opacity, 0.65);
  filter: var(--kendo-disabled-filter, grayscale(0.1));
  pointer-events: none;
  box-shadow: none;
}
```
css generated before migration
```
.k-disabled,
.k-widget[disabled],
.k-disabled {
  outline: none;
  cursor: default;
  color: var(--kendo-disabled-text, #8f8f8f);
  background-color: var(--kendo-disabled-bg, );
  border-color: var(--kendo-disabled-border, );
  opacity: var(--kendo-disabled-opacity, );
  filter: var(--kendo-disabled-filter, );
  pointer-events: none;
  box-shadow: none;
}
```
as you can see in the color attribute does not get any value, where before migration it was getting as of the variable $kendo-disabled-text
So there are two issues here,

- How we can implement different colors on variables that are the same now;
- In the case of $kendo-disabled-text  on-app-surface variable is not applied

**Comment by epetrow** [comment]
@molequetu, I can confirm that in terms of the disabled styling the aforementioned behaviour is a legitimate bug. We currently have an open [issue](https://github.com/telerik/kendo-themes/issues/5120) to which you can subscribe in order to track it's development. 

Touching on the first point that you mentioned, what you are currently trying to achieve will not work because the values assigned in `@import "../../athena/scss/base/_variables.scss";` are being overwritten in the `$kendo-colors` map which you are generating below. 

With the new Color system all color customizations are done through the `$kendo-colors` map as showcased in our docs ([customization](https://www.telerik.com/design-system/docs/themes/theme-bootstrap/customization/), [migration](https://www.telerik.com/design-system/docs/themes/theme-bootstrap/migration/)).

That said the Color system is backwards compatible and if you wish to preserve the "old" granular customization behaviour, you can do that by setting `$kendo-enable-color-system` to false. This is not recommended however as the new Color System provides a more predictable, consistent, and easier way to customize the colors in the theme.

**Comment by molequetu** [comment]
@epetrow OK for the issue with the disabled styling.

About the first point, so it seems that with the new color system there is no way to have different values for e,g $kendo-disabled-text and $kendo-body-text as now they are the same - prior to the new color system these variables could have different values as I mentioned.
